[CI] EsqlClientYamlIT test {p0=esql/46_downsample/Stats from downsampled and non-downsampled index simultaneously with implicit casting} failing

**Build Scans:**
- [elasticsearch-intake #33350 / 9.2.5_bwc-snapshots](https://gradle-enterprise.elastic.co/s/hvpgay2w2nnba)
- [elasticsearch-intake #33282 / 9.2.5_bwc-snapshots](https://gradle-enterprise.elastic.co/s/lgc2z3ao3h7t2)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v9.2.5#yamlRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT" -Dtests.method="test {p0=esql/46_downsample/Stats from downsampled and non-downsampled index simultaneously with implicit casting}" -Dtests.seed=32FB93D16DFDF530 -Dtests.bwc=true -Dtests.locale=hy-AM -Dtests.timezone=America/Mendoza -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Desql%2F46_downsample%2FStats%20from%20downsampled%20and%20non-downsampled%20index%20simultaneously%20with%20implicit%20casting%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.lang.AssertionError: field [values.0.0] is null
```

**Issue Reasons:**
- [main] 2 failures in test test {p0=esql/46_downsample/Stats from downsampled and non-downsampled index simultaneously with implicit casting} (0.4% fail rate in 461 executions)
- [main] 2 failures in step 9.2.5_bwc-snapshots (4.8% fail rate in 42 executions)
- [main] 2 failures in pipeline elasticsearch-intake (4.4% fail rate in 45 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

```
Caused by: org.elasticsearch.common.io.stream.NotSerializableExceptionWrapper: unsupported_operation_exception: Received an index (4) outside of range for AggregateMetricDoubleBlock.
```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -267,9 +267,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.heap_attack.HeapAttackSubqueryIT
   method: testManyRandomTextFieldsInSubqueryIntermediateResultsWithSortManyFields
   issue: https://github.com/elastic/elasticsearch/issues/140664
-- class: org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT
-  method: test {p0=esql/46_downsample/Stats from downsampled and non-downsampled index simultaneously with implicit casting}
-  issue: https://github.com/elastic/elasticsearch/issues/140774
 - class: org.elasticsearch.xpack.inference.integration.AuthorizationTaskExecutorIT
   method: testCreatesEisChatCompletion_DoesNotRemoveEndpointWhenNoLongerAuthorized
   issue: https://github.com/elastic/elasticsearch/issues/140802

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/46_downsample.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/46_downsample.yml
@@ -260,8 +260,8 @@ setup:
         - method: POST
           path: /_query
           parameters: []
-          capabilities: [aggregate_metric_double_v0, dense_vector_agg_metric_double_if_version]
-      reason: "Support for casting aggregate metric double implicitly when present in aggregations"
+          capabilities: [aggregate_metric_double_default_metric, dense_vector_agg_metric_double_if_version]
+      reason: "Requires default metric to run TS ... | STATS max(agg_metric)"
 
   - do:
       indices.downsample:

